src/core/drive/page_view.js

```diff
@@ -55,8 +55,17 @@ export class PageView extends View {
     return this.snapshotCache.get(location)
   }
 
+  shouldMorphPage(snapshot, visit) {
+    return snapshot.shouldMorphPage &&
+            (this.isPageRefresh(visit) || this.isReplaceMethodMorph(visit))
+  }
+
   isPageRefresh(visit) {
-    return !visit || (this.lastRenderedLocation.pathname === visit.location.pathname && visit.action === "replace")
+    return !visit || (this.lastRenderedLocation.href === visit.location.href && visit.action === "replace")
+  }
+
+  isReplaceMethodMorph(visit) {
+    return !visit || (visit.action === "replace" && visit.replaceMethod === "morph")
```

Comment:
`!visit` reads a bit weird here; is it needed?

Reply:
I'm thinking that `isReplaceMethodMorph(null) === true` reads wrong. The method is used together with `isPageRefresh` which would return true if there's no visit, so I don't think it makes sense to have here.

Reply:
Good call, fixing